### DISCUSSION TITLE
Automation editor redesign

### DISCUSSION THREAD

**Discussion by matthiasdebaat** [home-assistant/frontend#10064] [Configuration]
Ryanan shared some redesign ideas in [Discord](https://discord.com/channels/330944238910963714/867715211619074048/874857629749489696), resulting in a number of design iterations by a couple of people. This [prototype](https://www.figma.com/proto/pAHuoEApZp5iY9W0ZOPxD5/Home-Assistant-Automation?page-id=1011%3A4491&node-id=1013%3A4719&viewport=501%2C48%2C0.7&scaling=scale-down&starting-point-node-id=1013%3A4719&show-proto-sidebar=1) is the current state and in the [Figma design](https://www.figma.com/file/pAHuoEApZp5iY9W0ZOPxD5/Home-Assistant-Automation?node-id=1011%3A4491) you can find the work in progress. 

![New automation](https://user-images.githubusercontent.com/20203284/134165601-88ba71a4-1d54-4db8-a3ed-52d4652dd235.png)

Biggest changes are:

- Mobile first approach
- Using dialogs for adding `triggers`, `conditions` and `actions`
- Using `chips` to summarize a `trigger`, `condition` or `action`

What do you think of the design? Feel free to share your feedback in `Figma comments` or in this discussion.

**Comment by reubenbijl** [comment]
Looks great, the biggest issue I've personally found when using the automations interface is understanding what values are expected in changing to and from states. So, say for example a door, is it open, closed, 0 or 1, and what does the capitalisation look like? Do I need to use an entity property? I think being able to select from previous state values for an entity would be really useful.

**Reply by matthiasdebaat** [reply]
Great suggestion to show previous states. 
<img width="662" alt="Schermafbeelding 2021-10-26 om 14 35 35" src="https://user-images.githubusercontent.com/20203284/138879958-1d22a202-a3c5-40c5-abd1-b16b7e9c5610.png">

For your example about the door I would suggest in using `device`
<img width="656" alt="Schermafbeelding 2021-10-26 om 14 35 11" src="https://user-images.githubusercontent.com/20203284/138880130-58901d79-8f27-46c5-aac4-abf372a4193a.png">

**Comment by svh1985** [comment]
I like it, this can also make the nested conditions a lot cleaner. Also; I think it's also important to add some visual structure to the nested conditions (eg adding card bg-colors, lines) to better differentiate the different conditions and hierarchy. This should make working with complex nested conditions a lot easier.

**Reply by matthiasdebaat** [reply]
Thanks! Can you give a small example (sketch) of what you have in mind?

**Reply by svh1985** [reply]
I created a rough sketch which adds structure by adding hierarchy lines, borders and colours.


**Rationales**
- Coloured borders: will allow the user to recognise the type even if the "header" is scrolled out of the viewport
- Lines will add hierarchy
- Easier way to set AND / OR conditions

Let me know what you think!

![image](https://user-images.githubusercontent.com/5213210/139954388-d39caf98-0bab-4241-a148-3684467d863c.png)

**Reply by maxshcherbina** [reply]
I love this design! My main problem with the current automation editor is how easy it is to get lost in nested conditions/actions when youre working on complicated automations. Color coding like this would aleviate a lot of that headache. I would also love a way to add comments to automations that are not just in YAML, but this clearer flow might alleviate some of those problems.

**Reply by svh1985** [reply]
@matthiasdebaat what do you think?

**Reply by matthiasdebaat** [reply]
I like it! Currently I'm working on something different. If anybody wants to make a detailed design in Figma, feel free to do so!

**Reply by woodmj74** [reply]
I think that would really work well - as mentioned its really easy to get lost in the current UI especially with nested conditions (which are so useful). A simple change of losing the boxes and having some lines would be good start

**Reply by kentivar** [reply]
This looks awesome! It would be great if you vlick on the top level header in each choice and or condition, action headers an make it collapse :)

**Reply by kentivar** [reply]
Also, if we had the abillity to add comments to each choice, action or condition and not only a general comment on the whole automation :)

**Comment by sim-san** [comment]
I saw a library from google which is used to build small code snippets with visual blocks.

![grafik](https://user-images.githubusercontent.com/7436938/143440539-f1580b29-a684-4b7e-90b5-db7139184dc8.png)

> The Blockly library adds an editor to your app that represents coding concepts as interlocking blocks. It outputs syntactically correct code in the programming language of your choice. Custom blocks may be created to connect to your own application. 

https://developers.google.com/blockly/

Maybe this can be graphically customized and adapted to Home Assistant. I just wanted to share my thoughts

**Reply by matthiasdebaat** [reply]
Thanks for your suggestion! The fun thing is that Domoticz is using Blocky for their automation editor. It has its pro's and cons functional and technical. I like that it's easy to read an automation, but personally I like the original idea of Ryanan better.

**Comment by emufan** [comment]
> What do you think of the design? 

Oh no. Please not another (hide everything from the user behind several ..., submenus, etc.)-idea.

I really have to ask, if you are using HA in adjusting your HA, implementing automations, monitoring entities, ... on your own or only look for UI in terms of design and completely ignore the daily usability.

The one, who have setup their HA once and perhaps only look in their automations and configs once per month and then think, "oh yea, much prettier" can agree, but those, who are starting and learning or actively adjusting their HA, like and have to see, what options are there and not to search in the deepest menus and visual hidden option ... 

If your idea is only the/a display mode, go for it, but if the ... and sub menus and hidden entries are your idea for the edit mode as well, please stop thinking in this direction and put on the hat and always look on usability, fewer clicks, no hidden options, ...

If the approach is, to make it more clear, than go the direction svh1985 suggested, but do not hide anything from the UI, which you need to edit.

**Reply by jspanitz** [reply]
I really like the color coding ideas and the indenting proposed by @svh1985 but 100% agree with @reubenbijl in that knowing what is expected is most of the battle.

One thing I absolutely do not agree with is the request by Ryanan for a mobile 1st design.  I have to say how bad an idea it is to make your configuration interfaces work on mobile when mobile should focus on end user usability and configuration should be left to focus on the full blown desktop / notebook / tablet user.

**Reply by emufan** [reply]
Completely agree. Esp. as mobile (at least on ios) is not really usable for such tasks, because of limitations/problems in entity search, dropdowns, ... bugs in copy&paste in editor, etc. etc.

At the end of the day, you would have broken both options. Desktop broken (or lost a lot of advantages of the big space, keyboard, ...) because of mobile first and mobile still not usable, because of the given examples.

**Reply by Petro31** [reply]
I completely think you aren't seeing the whole picture here.  The example he posted is a simple trigger that takes

```
- platform: state
  entity_id: input_boolean.vacation
  to: 'off'
```
and turns it into editable chips.  I.e.  You change the entity_id and the word off without touching the yaml or in separate places.

![image](https://user-images.githubusercontent.com/35082313/147134604-5794f05d-1722-47f3-9e45-864f83dc1f3c.png)

The more you add to the initial trigger, the longer the phrase would be and more chips in it would exist.  THis works extremely well on desktop and on mobile and it displays ALL relevant information in a pleasant way.  No information is hidden.

e.g.
```
- platform: state
  entity_id: input_boolean.vacation
  to: 'off'
  from: 'on'
  for: "00:00:00"
```

would turn into 

When (Vacation) changes from ('on') to ('off') for ('00:00:00'). 

Literally the exact same information in a tight space, all editable without expanding or more clicks fully making your points... pointless.

**Reply by emufan** [reply]
No. I see and saw this "whole picture".

But can you then show/explain, where I can see what is possible to set. The from, the to, the attribute, the timeframe, the trigger types, the relevant settings of other trigger types, the ... Most probably everything hidden behind the ![image](https://user-images.githubusercontent.com/22775515/147137491-9f83180c-c75e-4bcb-bc48-88636fa82739.png).

That was, what I referred to. Yes, for display, I like this (beside the next question), but for setting up and see all possibilities, you will lose a lot of information or again have to click 10 times more often than before.

The next question: I would like to discuss a more complex mockup and not the simplest. With more (combines) trigger types, more triggers, trigger ids, and and or combinations, etc. I would wonder, if this will then be that clear and easy to read as in such simple scenario. 😉

**Reply by Petro31** [reply]
> No. I see and saw this "whole picture".
> 
> 
> 
> But can you then show/explain, where I can see what is possible to set. The from, the to, the attribute, the timeframe, the trigger types, the relevant settings of other trigger types, the ... Most probably everything hidden behind the ![image](https://user-images.githubusercontent.com/22775515/147137491-9f83180c-c75e-4bcb-bc48-88636fa82739.png).
> 
> 
> 
> That was, what I referred to. Yes, for display, I like this (beside the next question), but for setting up and see all possibilities, you will lose a lot of information or again have to click 10 times more often than before.
> 
> 
> 
> The next question: I would like to discuss a more complex mockup and not the simplest. With more (combines) trigger types, more triggers, trigger ids, and and or combinations, etc. I would wonder, if this will then be that clear and easy to read as in such simple scenario. 😉

Did you try the mock-up? Nothing is hidden in the "..." menu outside what's already there;  edit in ui, edit as yaml.  You click the add trigger button... just like it's done today in the current ui.

Edit:  actually, I'm pretty sure you didn't try it.  I just counted clicks and it's less clicks if you add any trigger except for device trigger.  That's the same number of clicks.  ~~So where are these extra clicks coming from?~~ Maybe from editing a trigger.

Edit 2:  It seems like it's 1 additional click when editing the trigger type to something else.  That could be solved by moving the edit button next to the "..." button instead of being inside the "..." button.  Making the entire UI functionally identical to the old UI.  Same number of clicks.  Same information.  More compact for mobile and will work well with desktop.

Heck, we could even pull the delete button out of that menu as well and make that faster.

**Reply by Petro31** [reply]
> The next question: I would like to discuss a more complex mockup and not the simplest. With more (combines) trigger types, more triggers, trigger ids, and and or combinations, etc. I would wonder, if this will then be that clear and easy to read as in such simple scenario. 😉

Each section will be compartmentalized like the current UI.  Also, what you're asking for is a nightmare in the current UI, so why should that matter if it's going to be optimized?

I think the big thing I notice with your statements is that you're just resistant to change.  Can you actually offer up suggestions to solve the issue at hand: Making the UI usable for BOTH mobile and desktop.  The current solution is not suitable for mobile, that's it.  Being resistant to change will just have your voice muffled because this change is necessary for a large group of the user base.  Advanced users always have YAML as a fallback, so without your input, that might be what you're left with.

So what can we (yes all of us) come up with to make this usable for **_all_** users; desktop, tablets, and mobile.

**Reply by jspanitz** [reply]
At the risk as being labeled resistant to change - this focus on mobile for automations is quite frankly absurd, The focus should be desktop and if mobile can be optimized as a result, great.

**Reply by spacegaier** [reply]
> At the risk as being labeled resistant to change - this focus on mobile for automations is quite frankly absurd, The focus should be desktop and if mobile can be optimized as a result, great.

Honest and serious question: Why "absurd" in your mind? Simply because you (presumably) edit HA on desktop? Then that would not be a valid reason IMHO.

As *petro* said, we should try to come up with something that works across all platforms.

**Reply by jspanitz** [reply]
Absurb in the sense that hampering the desktop UI to provide usabliity on mobile is a bad design choice. Unfortunately that's what was done here.  Very few multi platform apps provide the full functionality in their mobile platforms as they recognize the limitations. Works great for ainple tasks, not so much for advanced / complex tasks.

**Reply by emufan** [reply]
> I think the big thing I notice with your statements is that you're just resistant to change.

I think this is the problem. Everyone who has a different mind and opinion is stupid in your point of view and has to be insulted. Not that great approach in my mind.

I'm resistant to deterioration.

And in the above example, why should the advanced settings again hidden behind a "advanced setting" (again +1 click and hidden options) if there is even enough space on mobile? And regarding your

> The current solution is not suitable for mobile, that's it. 

Where is this 
![image](https://user-images.githubusercontent.com/22775515/147205782-3b8ea183-438a-42b3-aa6f-1613c03c89f9.png)

not usable on mobile? One type dropdown, some (conditional) lines everything directly reachable.

Compared to the new deterioration, several screens, still to scroll, several clicks and then hidden attributes, ... On on desktop you only use 10% of the screen. You speak about "on both", but this isn't "on both" anymore. Even not an improvement for mobile (see above).

And again, for these triggers is might work, but you spoke about the big picture and I think about the steps in the future. And if this will change, what about the other changes for conditions, esp. and and or combined conditions, selection, ... in the same UI style?

**Reply by Petro31** [reply]
@jspanitz  advanced users always have yaml as a fallback.  And to be honest, yaml with VSCode far surpasses the current UI hands down.

@emufan You still don't get it.  No one said your ideas are stupid.  The issue is you don't have an idea.  Your idea is resistance.  There's a problem at hand and you are unwilling to participate in solving that problem.  That's the problem.  Until you understand that, this argument is moot.

> not usable on mobile? One type dropdown, some (conditional) lines everything directly reachable.

No it's not usable.  Click the dropdown on mobile.  Make sure you're at the bottom of the page.  Now try scrolling.  Guess what?  It doesn't work well.  Now try that with your left hand and your right hand.  Notice how hard it is when the dropdown covers where your thumb should scroll?

The unfortunate truth is that power users aren't the only users.  There are 5 major media platforms, and one of the largest complaint from 3 of them is "Editing in the UI is impossible on mobile".  There are countless posts on Reddit, Facebook, and discord complaining about this.  You don't see it here or on the forums because this is where power users reside (for the most part).

So again, I ask you to help solve the problem not throw the problem away.  Look past the tip of your noses and put yourself in their shoes.  They have JUST AS MUCH RIGHT to this software as you do.

**Reply by emufan** [reply]
I still don't see the need, the requests or the statistics for this deterioration for desktop users.

> They have JUST AS MUCH RIGHT to this software as you do.

Who is THEY? How many are these THEY? And how this is balanced to the other THEY (which is the majority as far as I think), who then have to live with this changes for the minority?

Honestly: Is there a poll or a telematics evaluation or a request with thousands of votes to make these changes? Or is it only because a new UI designer is in da house. As said, UI is more than beautiful look, but esp. UX and usability. And this is not getting improved on desktop, e.g. on the new config menu or with this proposal in this thread.

Esp. a telematics, how many automations are getting created or changed on devices below let's say below 10''?

If the answer is only, "we don't care that the majority of users are on desktop for such tasks and only sometimes adjust or play around with single settings on mobile" (what is easly possible but already limited because of editor on iOS, ...), but nevertheless, we will do the deterioration for desktop users and don't use the desktop space anymore and introduce a lot of pages and hidden "advantage setting" menus (with further clicks) and those who liked the editing on desktop and now sees these deterioration should please go to yaml mode - as you said - I still miss the use case for this change.

Let's be crisp:
- Why the "advantage setting" dropdown on an empty page as in the mockup?
- And secondly, as asked before already, how should this design approach would work for the action areas, combined conditions, ...

**Reply by jspanitz** [reply]
It really does appear your view is the only view that's correct.  One wonders why people don't participate.  This is a UI discussion and now you throw in editing YAML.  Well I suppose you edit yaml in the UI, quite the stretch.  Ok well with that in mind mobile users can just get a larger device.

Again my objection is not to improving the UI but it IS to a mobile first approach (one of the 3 bullets that started this post).  I find it hard to believe that the majority of users are using mobile to perform these tasks.  My vote still goes to improving the product first, worrying about advanced editing on mobile second.

**Reply by jspanitz** [reply]
On another note I just to confirm, I went into automations on my mobile device, what is all the fuss about?  It literally looks exactly like it does on the desktop. Nothing gets cut off, the screen is fully used.  I just created an automation just like i would on a desktop with the same pain points. 

The issues with the automation UI are still one of color coding, formatting and indenting across all platforms.

**Reply by Petro31** [reply]
> Who is THEY? How many are these THEY? And how this is balanced to the other THEY (which is the majority as far as I think), who then have to live with this changes for the minority?

Other users.  Just spend time on facebook, discord, and reddit.  You will see these posts.  I'm over all of them because I moderate all of them.  I see all the complaints.  Power users aren't complaining.  Power users use YAML.  Desktop users are complaining about missing functionality in the UI.  Mobile users are complaining that it straight up doesn't work at all.

> Again my objection is not to improving the UI but it IS to a mobile first approach (one of the 3 bullets that started this post). I find it hard to believe that the majority of users are using mobile to perform these tasks. My vote still goes to improving the product first, worrying about advanced editing on mobile second.

No, it's not mobile first.  It's include them.  This is what I mean by looking past the tip of your nose.  It's ridiculous that you guys still have the "ME ME ME ME ME ME" mentality in this conversation.  It's not about you.  The current UI does not work well on all mobile platforms.  That's it.  There's nothing more to this.  It may work on yours just fine, but it doesn't work on all of them.

> It really does appear your view is the only view that's correct. One wonders why people don't participate. 

You aren't even giving ANY leeway in this.  It's your way or the high way.  We are asking for ideas and you are saying no don't change it.  Is that really a discussion?  No.  It's like talking to a brick wall and it's extremely frustrating.

>Why the "advantage setting" dropdown on an empty page as in the mockup?

Well this is a discussion, and that's just what @matthiasdebaat  came up with.  Contrary to your belief, you actually can offer up something different than that menu.  I know it's a hard concept to grasp, but you can do that.

I already mentioned that they could be made into buttons.  It could be made into chips.  It could be made into a separate window.

> And secondly, as asked before already, how should this design approach would work for the action areas, combined conditions, ...

I like what @svh1985 came up with as it makes the most sense for nesting & complexity.  Combine that with a chips phrase for the target field and service field and you have a usable ui.  The only thing that would need to be addressed is the data section, which most likely won't be something simple in design.

**Reply by Petro31** [reply]
Maybe I should Identify the goals of the redesign (maybe @matthiasdebaat can add some points to this list):

1. Reduce Y scrolling.
2. Remove problematic ui elements that mobile devices have trouble with.
3. Make something that works well for all platforms.

Now, lets hear your suggestions.

**Reply by jspanitz** [reply]
@Petro31 literally right here:
![image](https://user-images.githubusercontent.com/51183445/147248134-ff616d3c-1c47-4835-81fb-7df1a37e19da.png)

I use mobile 90% of the time just not for config changes - i'm all for improving it, just not making it the priority.  I love the color coding and the formatting changes proposed, just don't focus them on mobile due to some mobile limitation.  I also stated this previously:

![image](https://user-images.githubusercontent.com/51183445/147248666-f727f67a-2b2b-4d0a-b582-767ea01a2281.png)

If I'm off base on this, then it's only due to the stated goal i circled in the screenshot above.

**Reply by matthiasdebaat** [reply]
@emufan The way you communicate isn't inviting to respond. For the sake of making a better Home Assistant better product _for all of us_ I've tried to summarize your input:
- People who actively adjust their HA don’t want to have everything hidden behind several ..., submenus, etc. For example `Advanced settings`.
- People that uses HA less than once per month think that this concept is “oh yea, much prettier”.
- Go for the direction svh1985 suggested, but do not hide anything from the UI, which you need to edit.
- You don’t see the need for mobile first, because (at least on iOS) it’s not really usable for such tasks. Desktop is better because of the screen size and keyboard shortcuts.
- "Everyone who has a different mind and opinion is stupid (…) and has to be insulted."
- This concept is deterioration for desktop.
- Why change? Are there statistics? What about me?

Feedback to me personal
- Do you use Home Assistant? “Or do you only look for UI in terms of design and completely ignore the daily usability.”
- “Put on the hat and always look on usability, fewer clicks, no hidden options, ..”
- “Or is it only because a new UI designer is in da house.”

So first about me, so we can talk about Home Assistant again. Yes, I have been using Home Assistant for a few years including the YAML live. I'm a educated and experienced designer, so I know a lot about usability, accessibility, and perhaps the hardest part, knowing your users. No, I'm not changing anything because I'm a designer, but I'm addressing pain points that a group is going to improve on. That's it, please stop making it personal about me.

Home Assistant has a lot of user types, one of which you represent. As mentioned before, there are users who want to do everything on mobile. Whether this group is larger or smaller than the group that prefers desktop does not matter. The need exists. Mobile first is a handy trick to make things nice on mobile. That doesn't mean the desktop will deteriorate, and this automation editor concept doesn't even have a desktop design yet! About hiding, things are already hidden. You are can't automate on your Home Assistant landing page. The question is to make it logical, to give information just in time. That's a matter of taste and discussion.

**Reply by Petro31** [reply]
> @Petro31 literally right here: ![image](https://user-images.githubusercontent.com/51183445/147248134-ff616d3c-1c47-4835-81fb-7df1a37e19da.png)
> 
> I use mobile 90% of the time just not for config changes - i'm all for improving it, just not making it the priority. I love the color coding and the formatting changes proposed, just don't focus them on mobile due to some mobile limitation. I also stated this previously:
> 
> ![image](https://user-images.githubusercontent.com/51183445/147248666-f727f67a-2b2b-4d0a-b582-767ea01a2281.png)
> 
> If I'm off base on this, then it's only due to the stated goal i circled in the screenshot above.

Yes, I see that goal.  And maybe this is where we think differently.  When I see mobile first, I understand exactly what the goals are:  Make this work for mobile.  As a developer of 15+ years, that usually entails what I described above.  It's the idea that a keyboard and mouse don't exist and Y scrolling should be reduced as much as possible.  You can definitely make a UI that works well both for mobile and desktop.  For example, my home assistant configuration for lovelace is designed around using it on all platforms.  So I removed all dropdowns, text boxes expand the entire view or they are popups that ask for entry, everything else is a button.  The UI looks and feels great for both desktop and mobile.  That's all that were doing here.  "How can we take the existing UI and make it work for both platforms".

**Reply by emufan** [reply]
Before we discuss any further, you please align between:

> you can definitely make a UI that works well both for mobile and desktop. [...] How can we take the existing UI and make it work for both platforms

and / v.s.

> That doesn't mean the desktop will deteriorate, and this automation editor concept doesn't even have a desktop design yet!

Are we now talking about different views for mobile and desktop or about one.? If Matthias has written "doesn't even have a desktop design yet" I'm now lost, because what Petro has written is completely the opposite and what we have seen in the config menu, there is only one view on both devices classes. And even on mobile it is not better than before.

@matthiasdebaat Thanks for summerizing my posts. The answers will come as well?

- Are there statistics? 
- People who actively adjust their HA don’t want to have everything hidden behind several ..., submenus, etc. For example Advanced settings. Why?

I don't understand esp. the approach to hide everything behind another click. Even if the screen has enough space?

**Reply by Petro31** [reply]
@emufan did you actually read any of the responses?  You keep talking about this 'hide everything' and I've mentioned at least 3 times now that the discussion on that is open.

As for the his comment about desktop/mobile, he's referring to the Figma designs.  He saying he hasn't even shown what it would look like in desktop.  Can you just stop trying to twist words around to win an imaginary argument and actually constructively join the discussion or leave.  I don't care which one you do, but your current approach will not last.

**Reply by reubenbijl** [reply]
Hello, and Merry Christmas,

It’s great to see some engaged users of Home Assistant contributing to this thread. I’ve just been through trying to understand where this is currently at and I feel like the biggest theme is the disagreement of the feature goals. I’m not an active member of enough of the various Home Assistant communities but feel like there must be a better place for this kind of discussion?

To quote Paulus - “ You should not have to adapt to technology”. [Perfect Home Automation](https://www.home-assistant.io/blog/2016/01/19/perfect-home-automation/). To me one of the successes of Home Assistant has been the relentless desire to keep making it simple to use and the feature goals support this intent well. If only designing a complex feature that works well for every user, regardless of the technology they have at hand was so simple.

**Comment by emufan** [comment]
> As for the his comment about desktop/mobile, he's referring to the Figma designs. He saying he hasn't even shown what it would look like in desktop. 

Don't understand (perhaps of my English capabilities). If he writes, that there will be a desktop design and you now as well, that it is not clear how it will kook like on desktop, this is 100% a statement, that it will not look the same as on mobile. But currently this is the case (e.g. on config admin menu), isn't it?

But the "Please leave the discussion, if you don't like, what we plan" is again your already postet "Everyone who has a different mind and opinion is stupid (…) and has to be insulted.". 

I proposed constructively, that the advanced setting dropdown (ans 99% of other) hiding of settings or options (like the developer tools->state->set state), ... brings no benefit.

And I constructively asked what the basis for the ongoing task are, Is there a poll or a telematics evaluation, how many users is using this UI on mobile and on desktop to balance at least the changes accordingly.

I'm working in software development as well and there always ask the developer to listen to the majority and not to develop only because of their own mood and ideas.

**Reply by Petro31** [reply]
> Don't understand (perhaps of my English capabilities). If he writes, that there will be a desktop design and you now as well, that it is not clear how it will kook like on desktop, this is 100% a statement, that it will not look the same as on mobile. But currently this is the case (e.g. on config admin menu), isn't it?

No.  It will look different but similar.  He was implying that he hasn't mocked up the layout yet in his design program. 

> But the "Please leave the discussion, if you don't like, what we plan" is again your already postet "Everyone who has a different mind and opinion is stupid (…) and has to be insulted.".

More word twisting.   I gave you an ultimatum.  I did not say your idea was stupid.  I said join the conversation constructively or leave.

> And I constructively asked what the basis for the ongoing task are, Is there a poll or a telematics evaluation, how many users is using this UI on mobile and on desktop to balance at least the changes accordingly.

You've been given the reason.  **You aren't accepting it**.  Ask again and you'll be violating the COC due to trolling and harassment.  If or when that happens, only you will be removed.  Not @jspanitz, not anyone else in this thread.  Just you.

> I'm working in software development as well and there always ask the developer to listen to the majority and not to develop only because of their own mood and ideas.

Again, this is not happening.  You are claiming this is happening. I'm not going to repeat myself.

**Reply by emufan** [reply]
> You've been given the reason. **You aren't accepting it**. Ask again and you'll be violating the COC due to trolling and harassment. If or when that happens, only you will be removed.

Sorry, this is unbelievable. And only because of this, I replied once again, even with the risk to get as threatened banned because of other opinion.

The only reason I got, was "**Other users**. Just spend time on facebook, discord, and reddit.". Of course the desktop (configuration) users, who are happy with the status quo will not write that frequently "everything great" as other once perhaps complain. For me such basis cannot be a basis for such changes, we currently see.

I'm a mathematician and I live with statistical significance of voting, evaluations, etc. and I'm pretty sure that "other users" without counts, statistics, see who is using what (esp. configuration) on which platform is the opposite of building HA balanced to the usage.

**Reply by Petro31** [reply]
>Sorry, this is unbelievable. And only because of this, I replied once again, even with the risk to get as threatened banned because of other opinion.

No.  Let me be clear.  It's not because of your opinion.  It's the way you are addressing the situation.  Take @jspanitz for example.  He has been nothing but cordial in this exchange and he does not agree.  This is you and your attitude.  Period.

> I'm a mathematician and I live with statistical significance of voting, evaluations, etc. and I'm pretty sure that "other users" without counts, statistics, see who is using what (esp. configuration) on which platform is the opposite of building HA balanced to the usage.

You're asking for data that doesn't exist and you know it, and you're badgering about it.  The data we have are purely what we see on a day to day basis from other users.  I'm sorry but I'm not going to sit here with an excel sheet and mark down each person I come across that likes or dislikes the current UI.

You've voiced your displeasure in any change on the UI and you refuse to offer any suggestions to progress it.  So what's the point of being here if you have nothing else to add other than frustration and anger?

**Comment by emufan** [comment]
> You've voiced your displeasure in any change on the UI

Again **No**. Not in **any** change, only in deterioration of the current state in terms of introducing more levels, more click and hidden stuff only for the sake of change. Change what you want, make it more beautiful, but don't introduce more clicks and hidden information everywhere. 

And yes, it's not only about this feature here, but only a result, where the above already happened (e.g. config menu, developer tools). And before the reason is later: No one showed another opinion, the thread was open, ...

But this is now the last comment on this from me in this thread. Think you will get it or still try to argue away everyone, who is not a fan of this direction.

**Reply by spacegaier** [reply]
Yes please, let's stop that feud (that is turning more and more towards personal accusations / attacks) that is just running in circles and let's get back to actual content.

**Comment by balloob** [comment]
I've blocked emufan from participating for 7 days for ignoring directions by moderators and developers to be constructive and stop repeating his message. He failed to comply. His behavior is trolling and actively derailing any constructive feedback and discussions.

**Comment by rahaaas** [comment]
This may not be the right place to chip in... 

I’d really like to see collapsing elements in the editor. As a refugee from HomeSeer, it’s the one aspect of UX that I miss. 

I know it’s also been suggested (#11468 and #13156), but I think it would also work extremely effectively with the current direction laid out here. 

Once a choice is made in each section, it can reflect just the current setting when collapsed, making even complex automations significantly more readable / rearrangeable, especially on mobile. 

It’s also entirely optional, so the only disadvantage is development complexity, but even that should be scalable. 

EDIT: Looking at the designs in more detail, I guess this would mean removing as much of the space / furniture as possible, and just leaving the actual actions, which can then be uncollapsed to edit.
